WordPress/gutenberg
# [RNMobile] Update gesture handler and reanimated libraries

**Gutenberg Mobile PR:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/4615





## Description


This PR updates the following libraries:
- `react-native-gesture-handler` to version `2.2.0` ([forked repository PR](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/15))
- `react-native-reanimated` to version `2.2.4` ([forked repository PR](https://github.com/wordpress-mobile/react-native-reanimated/pull/11))

Additionally, due to API changes in the newer versions of both libraries, a set of modifications have been applied.

## Testing Instructions





### Test example
1. Apply the following patch diff that incorporates an example that uses the new APIs of the libraries:
<details><summary>Click here to display the patch diff</summary>

```diff
diff --git a/packages/edit-post/src/components/visual-editor/index.native.js b/packages/edit-post/src/components/visual-editor/index.native.js
index 886e56e493938b4b63c08199e39fc90f7accf4e5..25850c6b5cab4e0f2808064fdbd79c7c2481c0d7 100644
--- a/packages/edit-post/src/components/visual-editor/index.native.js
+++ b/packages/edit-post/src/components/visual-editor/index.native.js
@@ -6,12 +6,77 @@ import { BlockList } from '@wordpress/block-editor';
 /**
  * External dependencies
  */
-import { Keyboard } from 'react-native';
+import { Keyboard, StyleSheet, View } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, {
+	useSharedValue,
+	useAnimatedStyle,
+	withSpring,
+} from 'react-native-reanimated';
 /**
  * Internal dependencies
  */
 import Header from './header';
 
+const TestComponent = () => {
+	const ballStyles = StyleSheet.create( {
+		container: {
+			position: 'absolute',
+			bottom: 64,
+			right: 32,
+			zIndex: 10,
+		},
+		ball: {
+			width: 100,
+			height: 100,
+			borderRadius: 100,
+			backgroundColor: 'blue',
+		},
+	} );
+
+	const isPressed = useSharedValue( false );
+	const offset = useSharedValue( { x: 0, y: 0 } );
+	const animatedStyles = useAnimatedStyle( () => {
+		return {
+			transform: [
+				{ translateX: offset.value.x },
+				{ translateY: offset.value.y },
+				{ scale: withSpring( isPressed.value ? 1.2 : 1 ) },
+			],
+			backgroundColor: isPressed.value ? 'yellow' : 'blue',
+		};
+	} );
+
+	const start = useSharedValue( { x: 0, y: 0 } );
+	const gesture = Gesture.Pan()
+		.onBegin( () => {
+			isPressed.value = true;
+		} )
+		.onUpdate( ( e ) => {
+			offset.value = {
+				x: e.translationX + start.value.x,
+				y: e.translationY + start.value.y,
+			};
+		} )
+		.onEnd( () => {
+			start.value = {
+				x: offset.value.x,
+				y: offset.value.y,
+			};
+		} )
+		.onFinalize( () => {
+			isPressed.value = false;
+		} );
+
+	return (
+		<View style={ ballStyles.container }>
+			<GestureDetector gesture={ gesture }>
+				<Animated.View style={ [ ballStyles.ball, animatedStyles ] } />
+			</GestureDetector>
+		</View>
+	);
+};
+
 export default class VisualEditor extends Component {
 	constructor( props ) {
 		super( props );
@@ -58,11 +123,14 @@ export default class VisualEditor extends Component {
 		const { isAutoScrollEnabled } = this.state;
 
 		return (
-			<BlockList
-				header={ this.renderHeader }
-				safeAreaBottomInset={ safeAreaBottomInset }
-				autoScroll={ isAutoScrollEnabled }
-			/>
+			<>
+				<TestComponent />
+				<BlockList
+					header={ this.renderHeader }
+					safeAreaBottomInset={ safeAreaBottomInset }
+					autoScroll={ isAutoScrollEnabled }
+				/>
+			</>
 		);
 	}
 }
```

</details>

2. Open any post/page.
3. Observe that a blue ball is displayed in the right lower corner.
4. Observe that the ball can be dragged and dropped around the screen.

### Unit tests pass
1. Run `npm run native test` and observe that all tests pass.
1. Check that PR check `Unit Tests / Mobile` pass.

## Screenshots 
N/A

## Types of changes




Dependencies update

## Checklist:
